pkg/kubectl/cmd/apply_test.go

```diff
@@ -136,9 +147,16 @@ func annotateRuntimeObject(t *testing.T, originalObj, currentObj runtime.Object,
 	if currentAnnotations == nil {
 		currentAnnotations = make(map[string]string)
 	}
-	currentAnnotations[annotations.LastAppliedConfigAnnotation] = string(original)
+
+	originalAccessor.SetAnnotations(map[string]string{})
```

Comment:
ok I have added a conditional. 

Without this empty  annotation, the PATCH request does not become empty. 
In kubectl execution the PATH request is actually empty 
`Request Body: {"metadata":{"annotations":{}}}`
However in this unit test, the request differs. 
`{"metadata":{"annotations":{"kubectl.kubernetes.io/last-applied-configuration":"{\"apiVersion\":\"unit-test.test.com/v1\",\"key\":\"value\",\"kind\":\"Widget\",\"metadata\":{\"annotations\":{},\"name\":\"widget\",\"namespace\":\"test\"}}\n"}}}` 

I made a previous comment on this PR  about that and asked you whether you new the reason of it. 

In this  [commit](https://github.com/kubernetes/kubernetes/pull/41151/commits/ed1da06d7692f9c5a75f36f36edc6ee910110fbd) submit message I also show that bunch of other PATCH requests going to the API server also contain the empty annotation.